modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistFilter.java

```diff
@@ -42,8 +42,12 @@ public class BlacklistFilter implements IgnitePredicate<InetAddress> {
     public BlacklistFilter(Collection<String> blacklist) {
         matchers = new ArrayList<>(blacklist.size());
 
-        for (String pattern : blacklist)
+        for (String pattern : blacklist) {
+            if (pattern == null || pattern.isEmpty())
```

Comment:
Let's also verify that `pattern.trim()` is also not empty. Maybe we have a shortcut for this in `IgniteUtils`, so eventually we can write something like:
`if (isNotEmpty(pattern))
    matchers.add(new IPv4Matcher(pattern));`

Reply:
nice catch! fixed